tests/nested/manual/hybrid-tpm-fde-preinstall-check/task.yaml

```diff
@@ -134,3 +136,56 @@ execute: |
   # Check that the basic availability check that do not consider shim, grub and kernel still works
   remote.exec "sudo sed -i 's/^VERSION_ID=.*/VERSION_ID=25.04/' /etc/os-release"
   remote.exec "sudo snap debug api /v2/systems/classic 2>/dev/null" | gojq '.result["storage-encryption"].support' | MATCH "available"
+
+  # Revert back to working preinstall check
+  remote.exec "sudo cp /cdrom/EFI/boot/bootx64.efi.orig /cdrom/EFI/boot/bootx64.efi" # restore correct shim
+  remote.exec "sudo sed -i 's/^VERSION_ID=.*/VERSION_ID=25.10/' /etc/os-release"     # enable secboot preinstall check
+
+  # Check that encryption is again available for the targeted system
+  remote.exec "sudo snap debug api /v2/systems/classic 2>/dev/null" | gojq '.result["storage-encryption"].support' | MATCH "available"
+  remote.exec "sudo snap debug api /v2/systems/classic 2>/dev/null" | gojq '.result["storage-encryption"]."encryption-type"' | MATCH "cryptsetup"
+
+  # Restart snapd to clear preinstall context from cache
+  remote.exec "sudo systemctl restart snapd.service"
+
+  # Requesting a fix action when there is no preinstall context available should result in error
+  echo '{"action": "fix-encryption-support", "fix-action": ""}' | remote.exec "sudo snap debug api -X POST -H 'Content-Type: application/json' /v2/systems/classic 2>/dev/null" > result
+  cat <<EOF > result.reference
+  {
+    "result": {
+      "message": "cannot use check action without cached encryption information"
+    },
+    "status": "Internal Server Error",
+    "status-code": 500,
+    "type": "error"
+  }
+  EOF
+
+  diff result result.reference
+
+  # Run preinstall check to provide cached context
+  remote.exec "sudo snap debug api /v2/systems/classic 2>/dev/null" | gojq '.result["storage-encryption"].support' | MATCH "available"
+  remote.exec "sudo snap debug api /v2/systems/classic 2>/dev/null" | gojq '.result["storage-encryption"]."encryption-type"' | MATCH "cryptsetup"
+
+  # Requesting a fix action with preinstall context available should succeed
+  echo '{"action": "fix-encryption-support", "fix-action": ""}' |
```

Comment:
This is also not related to the spread test, but shouldn't the daemon fail for an empty `fix-action` or does an empty value have a special meaning?